src/snowflake/snowpark/file_operation.py

```diff
@@ -239,22 +239,27 @@ def put_stream(
         Returns:
             An object of :class:`PutResult` which represents the results of an uploaded file.
         """
+        stage_location = stage_location.strip()
+        if not stage_location:
+            raise ValueError("stage_location cannot be empty")
+        if stage_location[-1] == "/":
+            raise ValueError("stage_location should end with target filename")
```

Comment:
How about adding what's a correct value for stage_location after telling them what's wrong?
For instance, "stage_location cannot be empty. It must be a full stage path with prefix and file name..., like <the correct format>"